src/components/cc-logs-controller/cc-logs-controller.js

```diff
@@ -0,0 +1,504 @@
+      this._resolvedMetadataRenderers = this._resolveMetadataRenderers(this.metadataRenderers, this.metadataDisplay);
+    }
+  }
+
+  render () {
+    return html`
+      <div class="header-slot">
+        <slot name="header"></slot>
+      </div>
+
+      <cc-button
+        class="header-scroll-button"
+        .icon=${followIcon}
+        accessible-name="${i18n('cc-logs-controller.scroll-to-bottom')}"
+        hide-text
+        @cc-button:click=${this._onScrollToBottomButtonClick}
+      ></cc-button>
+
+      <cc-popover
+        class="header-options-popover"
+        .icon=${optionsIcon}
+        accessible-name="${i18n('cc-logs-controller.show-logs-options')}"
+        hide-text
+        position="bottom-right"
+      >
+        ${this._renderOptions()}
+      </cc-popover>
+
+      <cc-logs-beta
+        class="logs"
+        ${ref(this._logsRef)}
+        .dateDisplay=${this.dateDisplay}
+        .filter=${this.filter}
+        ?follow=${this.follow}
+        .limit=${this.limit}
+        .logs=${this.logs}
+        .metadataRenderers=${this._resolvedMetadataRenderers}
+        ?strip-ansi=${this.stripAnsi}
+        style=${PALETTES[this.palette]}
+        .timezone=${this.timezone}
+        ?wrap-lines=${this.wrapLines}
+      ></cc-logs-beta>
+    `;
+  }
+
+  _renderOptions () {
+    return html`
+      <div class="options">
+        ${this._renderDisplayOptions()}
+        ${this._renderDateOptions()}
+        ${this._renderMetadataOptions()}
+      </div>
+    `;
+  }
+
+  _renderDisplayOptions () {
+    const PALETTE_CHOICES = Object.entries(PALETTES)
+      .map(([p]) => ({ label: this._getPaletteLabel(p), value: p }));
+
+    return html`
+      <div class="options-header">
+        <cc-icon .icon=${displayIcon} size="lg"></cc-icon>
+        <span>${i18n('cc-logs-controller.option-header.display')}</span>
+      </div>
+      <div class="options-group">
+        <cc-select
+          label="${i18n('cc-logs-controller.palette')}"
+          .options=${PALETTE_CHOICES}
+          .value=${this.palette}
+          @cc-select:input=${this._onPaletteChange}
+        ></cc-select>
+
+        <label for="strip-ansi">
+          <input id="strip-ansi"
+                 type="checkbox"
+                 @change=${this._onStripAnsiChange}
+                 .checked=${this.stripAnsi}> ${i18n('cc-logs-controller.strip-ansi')}
+        </label>
+
+        <label for="wrap-lines">
+          <input id="wrap-lines"
+                 type="checkbox"
+                 @change=${this._onWrapLinesChange}
+                 .checked=${this.wrapLines}> ${i18n('cc-logs-controller.wrap-lines')}
+        </label>
+      </div>`;
+  }
+
+  _renderDateOptions () {
+    const DATE_DISPLAY_CHOICES = DATE_DISPLAYS
+      .map((d) => ({ label: this._getDateDisplayLabel(d), value: d }));
+    const TIMEZONE_CHOICES = TIMEZONES
+      .map((z) => ({ label: this._getTimezoneLabel(z), value: z }));
```

Comment:
question: can/should we move this to the beginning of the file (and do it just once)?

This also means removing the `_getDateDisplayLabel` and `_getTimezoneLabel`.

We do something similar here: https://github.com/CleverCloud/clever-components/blob/master/src/components/cc-pricing-estimation/cc-pricing-estimation.js#L22-L33

Reply:
I don't want to call the i18n function outside the instance of the component.